firedrake/preconditioners/pmg.py

```diff
@@ -103,44 +133,27 @@ def initialize(self, pc):
         pdm.setRefine(None)
         pdm.setCoarsen(self.coarsen)
         pdm.setCreateInterpolation(self.create_interpolation)
+        # We need this for p-FAS
+        pdm.setCreateInjection(self.create_injection)
+        pdm.setSNESJacobian(_SNESContext.form_jacobian)
+        pdm.setSNESFunction(_SNESContext.form_function)
+        pdm.setKSPComputeOperators(_SNESContext.compute_operators)
+
         pdm.setOptionsPrefix(pc.getOptionsPrefix() + "pmg_")
         set_function_space(pdm, get_function_space(odm))
 
         parent = get_parent(odm)
         assert parent is not None
-        add_hook(parent, setup=partial(push_parent, pdm, parent), teardown=partial(pop_parent, pdm, parent),
```

Comment:
Was it necessary to change these lines (for lint etc)?

Reply:
Not at all, it was just my obsesion to split lines at 77 characters.